### PR TITLE
Fixed some issues - fully described in the PR

### PR DESCRIPTION
**Pull request by amitamir**
1. BF removed/added/active/inactive/moved on a runner would not mark the runner as dirty
2. renamed GingerUIPage CurrentItem into CurrentItemToSave as some pages had CurrentItem property
3. fixed an issue where creating a new solution would throw an exception
4. Current Selected Item was wrongly assigned to the solution object when viewing the Variable page in BF View page/ Automate Page

Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is point to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for Existing Functionality
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments

### CODE REVIEW

File: Ginger/GingerCoreCommon/Repository/RepositoryItemBase.cs [hunk 1]
```diff
@@ -1072,6 +1074,10 @@ private void DirtyCheck(string name)
 
         internal void RaiseDirtyChanged(object sender, EventArgs e)
         {
+            if (!SourceDirtyQueue.Contains(sender as RepositoryItemBase))
+            {
+                SourceDirtyQueue.Enqueue(sender as RepositoryItemBase);
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/Repository/RepositoryItemBase.cs:1079]: I don't see SourceDirtyQueue is being used anywhere, do we need it?

File: Ginger/GingerCoreNET/WorkSpaceLib/SolutionOperations.cs [hunk 2]
```diff
@@ -200,9 +200,12 @@ public void SaveSolution(bool showWarning = true, Solution.eSolutionItemToSave s
                 Solution.RepositorySerializer.SaveToFile(Solution, Solution.FilePath);
                 Solution.SetDirtyStatusToNoChange();
                 Reporter.HideStatusMessage();
-                if (WorkSpace.Instance.SolutionRepository.ModifiedFiles.Contains(Solution))
+                if (WorkSpace.Instance.SolutionRepository != null)
                 {
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/WorkSpaceLib/SolutionOperations.cs:204]: merge this 2 if's together

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreCommon/Repository/RepositoryItemBase.cs (hunk 1 above)
Comment (amitamir) [reply Ginger/GingerCoreCommon/Repository/RepositoryItemBase.cs]: is part of displaying what was changed in the repository item, removed it as of now as there is some more work to do on that area

File: Ginger/GingerCoreNET/WorkSpaceLib/SolutionOperations.cs (hunk 2 above)
Comment (amitamir) [reply Ginger/GingerCoreNET/WorkSpaceLib/SolutionOperations.cs:204]: merged the 2 ifs into 1

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**